### PR TITLE
Kernel: misc fixes

### LINKED ISSUE #6948: QEMU Q35 ACPI tables might need special handle if overlapping two physical pages
Testing latest changes from master on qemu Q35 machine type, boot process halts at this stage:
```
[Kernel]: ACPI: Using RSDP @ P0x000f5a90
[Kernel]: ACPI: Main Description Table valid? true
[Kernel]: ACPI: Using RSDT, enumerating tables @ P0x1ffe21ac
[Kernel]: ACPI: RSDT revision 1, total length: 56
[Kernel]: ACPI: Initializing Fixed ACPI data
[Kernel]: ACPI: Searching for the Fixed ACPI Data Table
[Kernel]: ACPI: Fixed ACPI data, Revision 3, length: 244 bytes
[Kernel]: ACPI: DSDT P0x1ffe0040
[#0 Kernel]: KUBSAN: member call on null pointer of type 'struct Thread'
[#0 Kernel]: KUBSAN: at ../Kernel/Arch/i386/CPU.cpp, line 280, column: 51
[#0 Kernel]: 0xc04ed2da (next: 0xc0107ef4)
[#0 Kernel]: 0xc04ef14a (next: 0xc0107f74)
[#0 Kernel]: 0xc0573f39 (next: 0xc0108034)
[#0 Kernel]: 0xc056a9b0 (next: 0xc0108108)
[#0 Kernel]: 0xc06138de (next: 0xc0108128)
[#0 Kernel]: 0xc0613c50 (next: 0xc0108178)
[#0 Kernel]: 0xc060d0b5 (next: 0xc01081e8)
[#0 Kernel]: 0xc060d2f9 (next: 0xc0108238)
[#0 Kernel]: 0xc0663815 (next: 0xc0108298)
[#0 Kernel]: 0xc0100145 (next: 0x00000000)
[Kernel]: KERNEL PANIC! :^(
[Kernel]: UB is configured to be deadly.
[Kernel]: at ../Kernel/UBSanitizer.cpp:28 in void print_location(const Kernel::UBSanitizer::SourceLocation&)
[#0 Kernel]: 0xc036c708 (next: 0xc0107e54)
[#0 Kernel]: 0xc04ed34c (next: 0xc0107ef4)
[#0 Kernel]: 0xc04ef14a (next: 0xc0107f74)
[#0 Kernel]: 0xc0573f39 (next: 0xc0108034)
[#0 Kernel]: 0xc056a9b0 (next: 0xc0108108)
[#0 Kernel]: 0xc06138de (next: 0xc0108128)
[#0 Kernel]: 0xc0613c50 (next: 0xc0108178)
[#0 Kernel]: 0xc060d0b5 (next: 0xc01081e8)
[#0 Kernel]: 0xc060d2f9 (next: 0xc0108238)
[#0 Kernel]: 0xc0663815 (next: 0xc0108298)
[#0 Kernel]: 0xc0100145 (next: 0x00000000)
```

### LINKED ISSUE #7445: Kernel: Audio doesn't work
With the change applied from 72c097530f192611a60053b15706468888b0ae68 audio doesn't work anymore.

cc @supercomputer7 :)

### LINKED ISSUE #8331: ProcFS: Per-process inodes need to be lazily allocated
The new ProcFS design is a lot nicer to work with and extend but unfortunately it also does *way* too much allocation!

It's probaby okay to allocate the `/proc/PID` directory node when a process start, but everything else that goes into that directory needs to be lazily allocated, including the file descriptor nodes.

The old design was awful to work with but it had one strength: it did not allocate anything unnecessarily. We have to win some of this back :)

cc @supercomputer7

### PR DESCRIPTION
**Pull request by supercomputer7**
Based on #7509.

**Kernel/PCI: Move the PCI components as a subfolder to the Bus directory**
**Kernel/USB: Move the USB components as a subfolder to the Bus directory** 
This is merely a cleanup change, nothing about any technical advantage.

**Kernel/ProcFS: Clean dead processes properly** 
Clean all the zombies from #7509.

**Kernel/TypedMapping: Round up length with offset_in_page**
**Kernel/ACPI: Don't ask from TypedMapping to map 2 pages if unnecessary**
Completely fixes #6948 :)

**Kernel/ProcFS: Lazily allocate all sub components of a PID folder**
Fixes #8331 to some extent. Maybe we should consider to allocate more sub components in the ProcFS lazily, but for now, this is already a big improvement :)

**Revert "Revert "Revert "Kernel: Fix PCI layout of i440fx QEMU machine"""**

I **highly suspect** the reason it was not working before was actually fixed in 017c5fc, so the SB16 DMA can work even if we initialize another E1000 network adapter. If we actually fixed the problem (thanks to @gunnarbeutner patch!), we should ensure we don't regress the effort of that patch. If that's the case, we can close #7445 safely :)

### CODE REVIEW

**Comment by supercomputer7** [comment]
Let's stop the zombie apocalypse that went on (that was my fault) by properly killing dead processes and their objects in the kernel :sweat_smile:

File: Kernel/Bus/USB/UHCIController.cpp [hunk 1]
```diff
@@ -146,7 +146,7 @@ KResult ProcFSUSBBusFolder::traverse_as_directory(unsigned fsid, Function<bool(c
     ScopedSpinLock lock(m_lock);
     VERIFY(m_parent_folder);
     callback({ ".", { fsid, component_index() }, 0 });
-    callback({ "..", { fsid, m_parent_folder->component_index() }, 0 });
+    callback({ "..", { fsid, m_parent_folder.strong_ref()->component_index() }, 0 });
```
Comment (awesomekling) [inline Kernel/Bus/USB/UHCIController.cpp:149]: You don't need to use `strong_ref()` when dereferencing a `WeakPtr`.

File: Kernel/ACPI/Parser.cpp [hunk 2]
```diff
@@ -147,7 +147,7 @@ UNMAP_AFTER_INIT void Parser::init_fadt()
     VERIFY(!m_fadt.is_null());
 
     // FIXME: We need at least two pages for mapping, since we can be on the "edge" of one page...
```
Comment (IdanHo) [inline Kernel/ACPI/Parser.cpp:149]: How about removing this FIXME then? :)

Comment (supercomputer7) [reply Kernel/ACPI/Parser.cpp:149]: Oops, forgot about that!

**Comment by gunnarbeutner** [comment]
Audio works for me. :)

**Comment by supercomputer7** [comment]
> Audio works for me. :)

I tested it with a wave file I had already converted and it worked OK. Thanks for the help and let's merge it now :)
cc @awesomekling

**Pull request merged**